src/main/java/net/minecraftforge/client/IRenderHandler.java

```diff
@@ -19,14 +19,27 @@
 
 package net.minecraftforge.client;
 
+import com.mojang.blaze3d.matrix.MatrixStack;
 import net.minecraftforge.api.distmarker.Dist;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.world.ClientWorld;
 import net.minecraftforge.api.distmarker.OnlyIn;
 
+import javax.annotation.Nullable;
+
+// TODO 1.16 Split this apart for the various uses instead of stuffing everything into one interface.
 @FunctionalInterface
 public interface IRenderHandler
 {
     @OnlyIn(Dist.CLIENT)
+    @Deprecated // Use MatrixStack override, TODO 1.16 remove
     void render(int ticks, float partialTicks, ClientWorld world, Minecraft mc);
+
+    /**
+     * @param matrixStack The matrix stack. Can be null when called at sites where vanilla does not supply one.
+     */
+    @OnlyIn(Dist.CLIENT)
+    default void render(int time, float partialTicks, ClientWorld world, @Nullable MatrixStack matrixStack) {
```

Comment:
1) This breaks the @FunctionalInterface on certian compilers IIRC. As it only works with one method.
2) Why remove the MC argument? I understand you can get it from Minecraft.getInstane, but having it passed in seems like better practice.

Reply:
It won't, plenty of stdlib functional interfaces have lots of default methods. Just look at Comparator.